app/src/main/java/nu/yona/app/ui/pincode/BasePasscodeActivity.java

```diff
@@ -568,16 +567,14 @@ private void displayData()
 	protected void postOpenAppEvent()
 	{
 		User loggedInUser = getSharedAppDataState().getUser();
-		if (loggedInUser.getLinks() == null || loggedInUser.getLinks().getYonaPostOpenAppEvent() == null)
+		if (loggedInUser.getEmbedded() == null || loggedInUser.getEmbedded().getYonaDevices() == null)
```

Comment:
Please put this behavior inside the user entity, exposed through a method like ``isDeleted()``.

Next, I don't understand this check. When there is no ``embedded`` or no devices inside it, it could very well indicate that the mobile number isn't confirmed yet. In that case, the account still exists.

What are you doing when you get a 404 from the server? Can't you put an explicit marker ``isDeleted`` in the entity?

Reply:
We have added this null check to avoid a crash in a rare scenario. I don't remember that now.
Getting a 404 from server and this is different right?

Reply:
I can add `isDeleted` to the user entity with same check happening there.

Reply:
The server will not have ``_embedded`` when the mobile number is not confirmed.

I propose to add a status value to the ``user`` entity, with three values:
* ``ACTIVE`` In this case, the user entity was successfully retrieved from the server and the ``yona:messages`` link is available.
* ``NUMBER_NOT_CONFIRMED`` In this case, the user entity was successfully retrieved from the server but the ``yona:messages`` link is NOT available.
* ``NOT_REGISTERED`` In this case, there is either no user entity at all (initial situation) or the server returned a 404 when trying to fetch it.

If that is available, you can rewrite statements like this as follows:
```
		if (loggedInUser.getStatus() != UserStatus.ACTIVE)
```

Reply:
See also [APPDEV-1239](https://jira.yona.nu/browse/APPDEV-1239).

Reply:
`NOT_REGISTERED` when does this state occur? I am little confused. 
If server returns 404 it can be that either URL is wrong or either API is wrong. So in this case user entity itself will not be there.
Once user response is returned from server, app will instantiate the entity in application, then the `self` url returned from the server will guide to fetch the user details further. 
For some reason if api returns 404 what is the purpose of maintaining the state, as we cannot recover the data anyway.

Reply:
It's unclear to me how the app registers the state that there is no user account (you can get into that state through a fresh app installation and by removing your account). If that is tracked by not having the user entity, that's fine with me. Then you still have three states, but marked differently:

Two are reflected through the enum status value:
* ``ACTIVE`` In this case, the user entity was successfully retrieved from the server and the ``yona:messages`` link is available.
* ``NUMBER_NOT_CONFIRMED`` In this case, the user entity was successfully retrieved from the server but the ``yona:messages`` link is NOT available.

The third state ("not registered") is indicated by the fact that the logged in user is null.